cci_cameroon/analysis/clustering_rumours/rumour_clustering_template.py

```diff
@@ -0,0 +1,412 @@
+
+# %%
+# using leiden community detection algorithm on the dataset
+coms = algorithms.leiden(G)
+draw_communities_graph(G, generate_colors(coms), coms)
+
+# %%
+len(coms.communities)
+
+# %%
+for index in range(len(coms.communities)):
+    print(data.comment[coms.communities[index]].index)
+    print("######################NEW Group####")
+
+# %%
+list(
+    data.iloc[[158, 161, 162, 163, 164, 165, 166, 167, 168, 171]].comment
+)  # belief about mask wearing by children
+
+# %%
+list(
+    data.iloc[
+        [
+            89,
+            92,
+            94,
+            104,
+            108,
+            110,
+            115,
+            116,
+            117,
+            119,
+            120,
+            126,
+            136,
+            139,
+            140,
+            143,
+            153,
+            155,
+            156,
+            157,
+            159,
+            169,
+            173,
+            180,
+            183,
+            184,
+            185,
+            186,
+            187,
+            189,
+            192,
+            193,
+            197,
+            198,
+            199,
+            208,
+            215,
+            216,
+            218,
+            270,
+            291,
+        ]
+    ].comment
+)
+
+# %%
+list(
+    data.iloc[
+        [
+            0,
+            1,
+            3,
+            4,
+            7,
+            13,
+            18,
+            22,
+            23,
+            24,
+            28,
+            31,
+            33,
+            36,
+            38,
+            64,
+            67,
+            75,
+            79,
+            85,
+            145,
+            328,
+            330,
+            331,
+            341,
+            343,
+        ]
+    ].comment
+)
+
+# %%
+list(
+    data.iloc[
+        [
+            2,
+            27,
+            30,
+            32,
+            35,
+            37,
+            54,
+            55,
+            56,
+            57,
+            58,
+            59,
+            60,
+            61,
+            62,
+            63,
+            65,
+            66,
+            80,
+            83,
+            84,
+            338,
+        ]
+    ].comment
+)
+
+# %%
+## Using walktrap community detection algorithm on the data subset
+
+# %%
+comsw = algorithms.walktrap(G)
+comsw2 = algorithms.walktrap(G2)
+
+# %%
+draw_communities_graph(G2, generate_colors(comsw2), comsw2)
+
+# %%
+list(to_use2.comment[comsw2.communities[0]])
+
+# %%
+list(to_use2.comment[comsw2.communities[1]])
+
+# %%
+list(to_use2.comment[comsw2.communities[2]])
+
+# %%
+list(to_use2.comment[comsw2.communities[3]])
+
+# %%
+
+# %% [markdown]
+# ## General observations
+#
+# * The higher the number of neigbors in the graph, the bigger the clusters formed. Need to agree on an average number of comments that should come through before algorithm is run in order to choose a suitable value for the number of neighbors.
```

Comment:
On this point, I think the idea is that we would run the community detection algorthm each time new comments come through but run it on all comments in the pool of unlabelled comments? So, for example, you may have a pool of 500 unlabelled comments and 2 new ones come through, so you would run the algorithm on 502 comments in that case. Maybe a good idea is to test on different size datasets and evaluate the performance and then we could have a step that checks the number of comments in the dataset and assigns k accordingly?